fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(res => {
+        return res.json()
+          .then(data => {
+            if (res.ok) return { todos: data, error: null };
+            return { todos: [], error: data };
+          })
+          .then(result => {
+            this.setState({ ...result });
+          })
+          .catch(err => console.error(err));
+      });
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+  }
+
+  addTodo(newTodo) {
+    const myHeaders = new Headers();
+    myHeaders.append('Content-Type', 'application/json');
+    const init = {
+      method: 'POST',
+      headers: myHeaders,
+      body: JSON.stringify(newTodo)
+    };
+    fetch('/api/todos', init)
+      .then(res => {
+        return res.json()
+          .then(data => {
+            if (res.ok) return { todos: this.state.todos.concat(newTodo), error: null };
+            return { todos: [...this.state.todos], error: data };
+          })
+          .catch(err => console.error(err));
+      });
+    this.componentDidMount();
+    /**
+    * Use fetch to send a POST request to `/api/todos`.
+    * Then 😉, once the response JSON is received and parsed,
+    * add the created todo to the state array.
+    *
+    * Do not mutate the original state array, nor any objects within it.
+    * https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data
+    *
+    * TIP: Be sure to SERIALIZE the todo object in the body with JSON.stringify()
+    * and specify the "Content-Type" header as "application/json"
+    *
+    * TIP: Use Array.prototype.concat to create a new array containing the contents
+    * of the old array, plus the object returned by the server.
+    */
+  }
+
+  toggleCompleted(todoId) {
+    const match = this.state.todos.filter(value => value.todoId === todoId)[0];
+    const obj = { isCompleted: !match.isCompleted };
+    const init = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(obj)
+    };
+    fetch(`/api/todos/${todoId}`, init)
+      .then(res => {
+        return res.json()
+          .then(data => {
+            if (res.ok) return obj;
+            return { ...obj, error: data };
+          });
```

Comment:
You need to update your state with `setState` so that the UI reflects the changes in your back end. Do this without re-downloading the universe.